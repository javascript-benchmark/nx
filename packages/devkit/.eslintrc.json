{
  "extends": "../../.eslintrc",
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {
        "@typescript-eslint/explicit-module-boundary-types": ["error"],
        "@typescript-eslint/no-restricted-imports": [
          "error",
          {
            "paths": [
              "@nx/workspace",
              "@angular-devkit/core",
              "@angular-devkit/architect",
              "@angular-devkit/schematics"
            ],
            "patterns": [
              {
                "group": ["nx/**/*"],
                "message": "Use requireNx() from packages/devkit/nx.ts OR use a type import instead.",
                "allowTypeImports": true
              },
              {
                "group": ["@nx/devkit/**/*"],
                "message": "Use a relative import"
              }
            ]
          }
        ]
      }
    },
    {
      "files": ["*.spec.ts"],
      "rules": {
        "@typescript-eslint/no-restricted-imports": "off"
      }
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    },
    {
      "files": ["./package.json"],
      "parser": "jsonc-eslint-parser",
      "rules": {
        "@nx/nx-plugin-checks": "error"
      }
    }
  ]
}
