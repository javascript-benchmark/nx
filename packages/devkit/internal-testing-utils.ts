/* eslint-disable @typescript-eslint/no-restricted-imports */
export * from 'nx/internal-testing-utils/assert-valid-migrations';
export * from 'nx/internal-testing-utils/run-migration-against-this-workspace';
export * from 'nx/internal-testing-utils/with-environment';
