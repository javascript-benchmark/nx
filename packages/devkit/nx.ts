// After Nx v18, this can be removed and replaced with either:
// - import {} from 'nx/src/devkit-exports'
// - import {} from 'nx/src/devkit-internals'
export function requireNx(): typeof import('nx/src/devkit-exports') &
  Partial<typeof import('nx/src/devkit-internals')> {
  try {
    let result = { ...require('nx/src/devkit-exports') };
    try {
      result = {
        ...result,
        // Remove in Nx v18, devkit should not support Nx v16.0.2 at that point.
        ...require('nx/src/devkit-internals'),
      };
    } catch {}
    return result;
  } catch {
    // Remove in Nx V17, devkit should not support Nx < 16 at that point.
    return require('./nx-reexports-pre16');
  }
}
