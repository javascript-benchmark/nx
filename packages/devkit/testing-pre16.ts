/* eslint-disable @typescript-eslint/no-restricted-imports */
export {
  createTreeWithEmptyWorkspace,
  createTreeWithEmptyV1Workspace,
} from 'nx/src/generators/testing-utils/create-tree-with-empty-workspace';
export { createTree } from 'nx/src/generators/testing-utils/create-tree';
