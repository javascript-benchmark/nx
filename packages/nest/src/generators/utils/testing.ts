import { addProjectConfiguration, Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';

export function createTreeWithNestApplication(appName: string): Tree {
  const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
  addProjectConfiguration(tree, appName, {
    root: `apps/${appName}`,
    sourceRoot: `apps/${appName}/src`,
    projectType: 'application',
    targets: {},
  });
  return tree;
}
