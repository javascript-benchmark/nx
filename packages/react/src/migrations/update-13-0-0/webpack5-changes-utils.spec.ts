import { addProjectConfiguration, Tree, updateJson } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { allReactProjectsWithStorybookConfiguration } from './webpack5-changes-utils';

describe('webpack5ChangesUtils', () => {
  let tree: Tree;
  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
  });

  it('should get project name and storybook configuration for all react projects', () => {
    addProjectConfiguration(tree, 'test-one', {
      root: 'libs/test-one',
      targets: {
        storybook: {
          options: {
            uiFramework: '@storybook/react',
            config: {
              configFolder: 'libs/test-one/.storybook',
            },
          },
        },
      },
    });
    addProjectConfiguration(tree, 'test-two', {
      root: 'libs/test-two',
      targets: {
        storybook: {
          options: {
            uiFramework: '@storybook/react',
            config: {
              configFolder: 'libs/test-two/.storybook',
            },
          },
        },
      },
    });
    const allReactProjects = allReactProjectsWithStorybookConfiguration(tree);

    expect(allReactProjects).toMatchSnapshot();
  });

  it('should ignore non-react projects with storybook configuration', () => {
    addProjectConfiguration(tree, 'test-one', {
      root: 'libs/test-one',
      targets: {
        storybook: {
          options: {
            uiFramework: '@storybook/angular',
            config: {
              configFolder: 'libs/test-one/.storybook',
            },
          },
        },
      },
    });

    const allReactProjects = allReactProjectsWithStorybookConfiguration(tree);
    expect(allReactProjects.length).toBe(0);
  });
});
