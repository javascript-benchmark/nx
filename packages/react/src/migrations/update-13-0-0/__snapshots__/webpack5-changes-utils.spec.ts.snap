// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack5ChangesUtils should get project name and storybook configuration for all react projects 1`] = `
[
  {
    "projectName": "test-one",
    "storybookConfigPath": "libs/test-one/.storybook",
  },
  {
    "projectName": "test-two",
    "storybookConfigPath": "libs/test-two/.storybook",
  },
]
`;
